Marlin/src/module/stepper_indirection.cpp

```diff
@@ -605,44 +599,57 @@ void reset_stepper_drivers() {
     L6470_init_to_defaults();
   #endif
 
+  #if STEALTHCHOP_ENABLED
+    bool stealthChop_by_axis[] = {false, false, false};
+    #if ENABLED(XY_STEALTHCHOP)
+      stealthChop_by_axis[0] = true;
+    #endif
+    #if ENABLED(Z_STEALTHCHOP)
+      stealthChop_by_axis[1] = true;
+    #endif
+    #if ENABLED(E_STEALTHCHOP)
+      stealthChop_by_axis[2] = true;
+    #endif
+  #endif
+
   #if AXIS_IS_TMC(X)
-    _TMC_INIT(X, planner.settings.axis_steps_per_mm[X_AXIS]);
+    _TMC_INIT(X, X_AXIS, 0);
   #endif
   #if AXIS_IS_TMC(X2)
-    _TMC_INIT(X2, planner.settings.axis_steps_per_mm[X_AXIS]);
+    _TMC_INIT(X2, X_AXIS, 0);
   #endif
   #if AXIS_IS_TMC(Y)
-    _TMC_INIT(Y, planner.settings.axis_steps_per_mm[Y_AXIS]);
+    _TMC_INIT(Y, Y_AXIS, 0);
   #endif
   #if AXIS_IS_TMC(Y2)
-    _TMC_INIT(Y2, planner.settings.axis_steps_per_mm[Y_AXIS]);
+    _TMC_INIT(Y2, Y_AXIS, 0);
   #endif
   #if AXIS_IS_TMC(Z)
-    _TMC_INIT(Z, planner.settings.axis_steps_per_mm[Z_AXIS]);
+    _TMC_INIT(Z, Z_AXIS, 1);
   #endif
   #if AXIS_IS_TMC(Z2)
-    _TMC_INIT(Z2, planner.settings.axis_steps_per_mm[Z_AXIS]);
+    _TMC_INIT(Z2, Z_AXIS, 1);
   #endif
   #if AXIS_IS_TMC(Z3)
-    _TMC_INIT(Z3, planner.settings.axis_steps_per_mm[Z_AXIS]);
+    _TMC_INIT(Z3, Z_AXIS, 1);
   #endif
   #if AXIS_IS_TMC(E0)
-    _TMC_INIT(E0, planner.settings.axis_steps_per_mm[E_AXIS_N(0)]);
+    _TMC_INIT(E0, E_AXIS, 2);
   #endif
   #if AXIS_IS_TMC(E1)
-    _TMC_INIT(E1, planner.settings.axis_steps_per_mm[E_AXIS_N(1)]);
+    { constexpr uint8_t extruder = 1; _TMC_INIT(E1, E_AXIS_N, 2); UNUSED(extruder); }
```

Comment:
These introduce a regression. `E_AXIS_N` now takes an argument. The `E_AXIS_N(1)` macro is correct.